src/shaka_scheme/system/gc/GC.hpp

```diff
@@ -0,0 +1,38 @@
+//
+// Created by Brandon Wong on 3/26/18.
+//
+
+#ifndef SHAKA_SCHEME_GC_HPP
+#define SHAKA_SCHEME_GC_HPP
+
+#include "shaka_scheme/system/gc/GCList.hpp"
+#include "shaka_scheme/system/gc/GCData.hpp"
+#include "shaka_scheme/system/base/Data.hpp"
+
+namespace shaka {
+
+    namespace gc {
+
+    /**
+         * @brief Implements the factory method of the GC, which allows the
+         * construction of GCData and automatically adds those GCData to the
+         * GCList
+         */
+        class GC {
+        public:
+
+            GC();
+            ~GC();
+            GC(const GC& other) = delete;
+            GC(GC&& other);
+
+            GCData *create_data(const shaka::Data& data);
```

Comment:
Isn't `GCNode` the placement for `shaka::NodePtr` (which is currently `std::shared_ptr<shaka::Data>`)? I know that we said that `GCNode` will be the type that is returned by `create_node` once we bind a `GC&` to a lambda that is `create_node` -- can you just state again why `GCData*` is being exposed at all here? The point is to not allow external users to `delete` the exposed raw pointer.

Reply:
@btwooton can verify me on this, but I believe `create_data` and `create_node` are different from each other in that `create_data` allows us to safely allocate GCData objects on the heap without having to worry about doing it in the Environment while passing GCList as an input parameter. `create_node` will be the next step in the GC implementation which will be designed in GCNode to substitute as an alias for NodePtr.

Reply:
create_data is not going to be exposed to the rest of the system. If it is preferred, we can find a way to make it a private method, but still be usable by create_node. Just know that this method will be called internally by the new create_node, which will return a GCNode, not a GCData.